packages/docusaurus-theme-mermaid/src/theme/Mermaid/index.tsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useEffect, useState, useRef} from 'react';
+import useIsBrowser from '@docusaurus/useIsBrowser';
+import mermaid from 'mermaid';
+import type {Props} from '@theme/Mermaid';
+
+export default function Mermaid({value}: Props): JSX.Element {
+  // When theme updates, rerender the SVG.
+  const [svg, setSvg] = useState('');
+  const isBrowser = useIsBrowser();
+  // Assign a unique ID to each mermaid svg as per requirements of
+  // `mermaid.render`.
```

Comment:
This looks very unsafe to have every effect mutate something... Effects aren't guaranteed to run in any order or run at any particular time, so this is practically not much better than assigning a random number to each SVG. This should be fixable with React 18's `useId` hook?

Reply:
I think so, shouldn't be an issue for now though 🤞

Reply:
![CleanShot 2022-10-07 at 18 14 02@2x](https://user-images.githubusercontent.com/749374/194600078-dcddcbef-5927-408c-b3e4-fc90482a26cc.png)

All diagrams on the same page will have the exact same ID because the counter is distinct for each instance.

I don't understand why Mermaid want an ID, can you explain?

If it's only used client-side, you could just wrap the component in `<BrowserOnly>` and generate a random number for each instance? (ie no need to increment)